manual/source/Tutorial/Algorithms/PatternMatching/OptimalSearchSchemes.rst

```diff
@@ -0,0 +1,65 @@
+.. sidebar:: ToC
+
+    .. contents::
+
+.. _tutorial-algorithms-optimal-search-schemes:
+
+Optimal Search Schemes
+======================
+
+Learning Objective
+  In this tutorial you will learn how to search a known pattern in a string or :dox:`StringSet` using a bidirectional index. The implemented algorithm in SeqAn is based on Optimal Search Schemes from the `FAMOUS paper <https://arxiv.org/abs/1711.02035>`_ and allows for searching for up to 4 errors based on Hamming distance or Edit distance.
+
+Difficulty
+  Average
+
+Duration
+  15 min
+
+Prerequisites
+  :ref:`tutorial-datastructures-sequences`, :ref:`tutorial-datastructures-indices`
+
+Overview
+--------
+
+Searching in an index with more than two errors is computationally very expensive and thus trivial approaches such as simple backtracking is not recommended.
+Optimal Search Schemes make use of a bidirectional index, splits the pattern to be searched into multiple peaces and enumerates it in a more efficient fashion.
+To use this algorithm, you can simply call the ``find()`` method.
+
+.. includefrags:: demos/tutorial/indices/find2_index_approx.cpp
+      :fragment: SinglePattern
+
+The first argument is the delegate function that will be called if a match of the pattern is found.
+It gets an iterator of the bidirectional index as only parameter and you can choose how to proceed with the hits.
+
+.. includefrags:: demos/tutorial/indices/find2_index_approx.cpp
+      :fragment: Delegate
+
+The second argument has to be a bidirectional index, currently SeqAn offers only bidirectional FM indices.
+(Please check the corresponding tutorial on :ref:`tutorial-datastructures-indices-fm-index` for more information how FM indices can be configured and constructed.)
+
+The third argument is a :dox:`String` that you want to search.
+
+The number of allowed errors (lower and upper bounds) are passed as template arguments.
+Please note that these parameters have to be `constexpr`.
+The distance metric is passed as fourth argument. You can choose between `HammingDistance()` and `EditDistance()` (also known as Levenshtein distance).
+
+You also have to possibility to pass a :dox:`StringSet` of patterns to search instead of a single :dox:`String`.
+The search can then be parallized by passing a fifth parameter tag `Parallel()` instead of `Serial()`.
+
+.. includefrags:: demos/tutorial/indices/find2_index_approx.cpp
+      :fragment: MultiplePatterns
+
+.. warning::
+
+      Please be aware that the same hits might be reported multiple times and you might have to filter these duplicated depending on your application, especially with larger errors numbers.
+
+.. note::
+
+      When searching a StringSet in parallel mode the delegate is a critical section, i.e. it might be accessed by different threads at the same time.
+      You have to take care that the threads do not interfer by yourself, e.g. write not into a variable at the same time.
```

Comment:
interfer**e**

This doesn't make sense. If it is marked as a critical section it is guaranteed that they don't interfere, right ❓